examples/dora_finetuning/dora_finetuning.py

```diff
@@ -0,0 +1,185 @@
+    tokenizer.pad_token = tokenizer.eos_token
+
+
+    # Load the dataset
+    dataset = load_dataset(data_path)
+
+    def tokenize_function(examples):
+        inputs = tokenizer(examples['text'], padding="max_length", truncation=True, max_length=cutoff_len)
+        inputs['labels'] = inputs['input_ids'].copy()  # setting labels for a language modeling task
+        return inputs
+
+    # Tokenize the dataset and prepare for training
+    tokenized_datasets = dataset.map(tokenize_function, batched=True, remove_columns=dataset['train'].column_names)
+
+    # Data collator to dynamically pad the batched examples
+    data_collator = DataCollatorWithPadding(tokenizer)
+
+    # Define training arguments
+    training_args = TrainingArguments(
+        output_dir=output_dir,
+        num_train_epochs=num_epochs,
+        per_device_train_batch_size=batch_size,
+        per_device_eval_batch_size=batch_size,
+        warmup_steps=100,
+        weight_decay=0.01,
+        logging_dir="./logs",
+        logging_steps=eval_step,
+        save_steps=save_step,
+        save_total_limit=2,
+        push_to_hub=push_to_hub,
+        hub_model_id=hub_model_id,
+        gradient_accumulation_steps=16,
+        fp16=True,
+        learning_rate=learning_rate,
+        hub_token=hf_token,
+    )
+
+    # Clear CUDA cache to free memory
+    torch.cuda.empty_cache()
+
+    # Initialize the Trainer
+    trainer = Trainer(
+        model=model,
+        args=training_args,
+        train_dataset=tokenized_datasets['train'],
+        eval_dataset=tokenized_datasets['test'],
+        data_collator=data_collator,
+    )
+
+    # Start model training
+    trainer.train()
+
+    # Save and push the trained model and tokenizer
+    if push_to_hub:
+        # Push the main model to the hub
+        trainer.push_to_hub(commit_message="Fine-tuned model")
+
+    # Save the model and tokenizer locally
+    model.save_pretrained(output_dir)
+    tokenizer.save_pretrained(output_dir)
+
+    # # Save and push the LoRA adapter if used
+    # if use_peft:
+    #     adapter_dir = os.path.join(output_dir, 'adapter')
+    #     model.save_pretrained(adapter_dir)
+    #     adapter = PeftModel(model)
+    #     adapter.push_to_hub(f"{hub_model_id}-adapter", use_temp_dir=False, commit_message="LoRA adapter")
+
+if __name__ == "__main__":
+    import argparse
+    parser = argparse.ArgumentParser(description="Fine-tune LLaMA with DoRA and PEFT")
+    parser.add_argument("--base_model", type=str, default="huggyllama/llama-7b", help="Base model path or name")
+    parser.add_argument("--data_path", type=str, default="timdettmers/openassistant-guanaco", help="Dataset path or name")
+    parser.add_argument("--output_dir", type=str, default="path/to/output", help="Output directory for the fine-tuned model")
+    parser.add_argument("--batch_size", type=int, default=1, help="Batch size")
+    parser.add_argument("--use_compile", default=False, help="Use Compile")
```

Comment:
I  removed use_compile from both the .py and the readme files, I ran into same index error, I got surprised since according to this [torch.compile](https://huggingface.co/docs/peft/main/en/developer_guides/torch_compile) document on HF torch.compile() should perfectly work for the Lora + Dora adapter setup.  🤔
For now, use_compile is completely removed but I am wondering to kindly know your idea on "why" they do not work well together?